src/Particular.LicensingComponent.UnitTests/Infrastructure/ThroughputCollectorTestFixture.cs

```diff
@@ -16,21 +17,24 @@ abstract class ThroughputCollectorTestFixture
         [SetUp]
         public virtual Task Setup()
         {
-            configuration = new ThroughputTestsConfiguration();
+            var testMethod = GetType().GetMethod(TestContext.CurrentContext.Test.MethodName);
+            var attribute = testMethod?.GetCustomAttributes(typeof(UseNonBrokerTransportAttribute), false).FirstOrDefault();
 
-            return configuration.Configure(SetThroughputSettings, SetExtraDependencies);
+            return configuration.Configure(SetThroughputSettings, SetExtraDependencies, attribute is not UseNonBrokerTransportAttribute);
         }
 
         [TearDown]
-        public virtual Task Cleanup()
-        {
-            return configuration?.Cleanup();
-        }
+        public virtual Task Cleanup() => configuration?.Cleanup();
 
         protected ILicensingDataStore DataStore => configuration.LicensingDataStore;
 
         protected IThroughputCollector ThroughputCollector => configuration.ThroughputCollector;
 
-        protected ThroughputTestsConfiguration configuration;
+        protected ThroughputTestsConfiguration configuration = new();
+
+        [AttributeUsage(AttributeTargets.Method)]
+        public class UseNonBrokerTransportAttribute : Attribute
```

Comment:
Feels a bit off to introduce this attribute to customise the setup of the test, I think refactoring the test may be simpler

Reply:
I felt the attribute was preferable rather than splitting related tests across different test fixtures.